classes/local/report/objectfs_report.php

```diff
@@ -97,14 +108,18 @@ public static function get_last_generate_status_report_runtime() {
     }
 
     public static function generate_status_report() {
+        $reportstarted = time();
         $reporttypes = self::get_report_types();
 
         foreach ($reporttypes as $reporttype) {
             $reportbuilderclass = "tool_objectfs\\local\\report\\{$reporttype}_report_builder";
             $reportbuilder = new $reportbuilderclass();
-            $report = $reportbuilder->build_report();
+            $report = $reportbuilder->build_report($reportstarted);
             objectfs_report_builder::save_report_to_database($report);
         }
+        // Throttle here for one second to make sure the snapshots have different
+        // $reportstarted if the report was called twice in a row and it was super fast.
+        sleep(1);
```

Comment:
This is a smell the design of the db table is not ideal

Reply:
Re-worked th edb schema, so there is no need to sleep now